### PR TITLE
Changes for using MPAS-Model V8.2.1 in Workflow

### PR DESCRIPTION
**Pull request by mos3r3n**
Using invariant file instead of initial file in the 2-stream IO;
Using new initial conditions and first background that use the "MODIFIED_IGBP_MODIS_NOAH" land use and "GMTED2010" topo;
Bug fixes of 60-3-km experiments to use the same physical parameterization;

This PR is jointly contributed by @ibanos90, @jim-p-w, and @mos3r3n 

Testing:
Tested on 1-month cycling 30-km/60-km and 60-3km/30km dual-resolution cycling hybrid 3DEnVar.
LETKF and LGETKF were not tested because the initial ensembles were generated using V7.x.

#### Tier 1:
 - [x] 3dvar_OIE120km_WarmStart
 - [x] 3denvar_OIE120km_IAU_WarmStart
 - [x]  3dvar_OIE120km_ColdStart
 - [x] 3dvar_O30kmIE60km_ColdStart
 - [x] 3denvar_O30kmIE60km_WarmStart
 - [x] 4denvar_OIE120km_WarmStart_TEST
 - [x] 4dhybrid_OIE120km_WarmStart_TEST
 - [x] eda_OIE120km_WarmStart
 - [x] letkf_OIE120km_WarmStart
 - [x] getkf_OIE120km_WarmStart
 - [x] ForecastFromGFSAnalysesMPT

### CODE REVIEW

**Comment by ibanos90** [comment]
@mos3r3n, can you add what tests did you run?

**Comment by mos3r3n** [comment]
> @mos3r3n, can you add what tests did you run?

Already added. Thanks!

**Comment by junmeiban** [comment]
@mos3r3n , scenarios/ForecastFromGFSAnalyses.yaml does not work.

**Comment by mos3r3n** [comment]
> @mos3r3n , scenarios/ForecastFromGFSAnalyses.yaml does not work.

Thanks for pointing it out. It was because I deleted the mem in the invariantstream.yaml but forgot to modify initialize/suites/ForecastFromExternalAnalyses.py and initialize/suites/CloudDirectInsertion.py. Now they have been modified.  I am testing it.

File: initialize/applications/InitIC.py [hunk 1]
```diff
@@ -64,24 +64,29 @@ def export(self, dtOffsets:list=[0]):
...
           meshNames.append(mesh.name)
           meshNCells.append(mesh.nCells)
+          meshRatios.append(mesh.meshRatio)
 
       zeroHR = '-0hr'
       queue = 'ConvertExternalAnalyses'
       subqueues.append(queue)
-      for (typ, meshName, nCells) in zip(meshTypes, meshNames, meshNCells):
+      for (typ, meshName, nCells, meshRatio) in zip(meshTypes, meshNames, meshNCells, meshRatios):
+        print(typ)
```
Comment (ibanos90) [inline initialize/applications/InitIC.py]: Can remove this print

File: test/testinput/3denvar_O30kmIE60km_WarmStart.yaml [hunk 2]
```diff
@@ -46,9 +46,15 @@ variational:
...
+  IRVISlandCoeff: IGBP
 workflow:
   first cycle point: 20180414T18
   # Note: only run for 1 cycle to reduce cost
   # Also required with only 15 inner iterations to avoid NaN in forecast
   # Can cycle for a full month when nInnerIterations==[60,60]
   final cycle point: 20180415T00
+
+verifymodel:
+  script directory: /glade/work/ivette/pandac/graphics/Graphics_Python_noPressure-noTemperature
```
Comment (ibanos90) [inline test/testinput/3denvar_O30kmIE60km_WarmStart.yaml]: I think you can remove these

File: test/testinput/3denvar_O30kmIE60km_WarmStart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 experiment:
-  name: '3denvar_O30kmIE60km_WarmStart_TEST'
+  name: '3denvar_O30kmIE60km_WarmStart_TEST_Modelv8.2.1_NEWINIT-1stBKG_files_verifyFixed'
```
Comment (ibanos90) [inline test/testinput/3denvar_O30kmIE60km_WarmStart.yaml]: I think you can keep the original exp name

File: scenarios/3dhybrid_O60-3kmIE60km_SpecifiedEnsemble_VarBC_v8.1.yaml
Comment (ibanos90) [inline scenarios/3dhybrid_O60-3kmIE60km_SpecifiedEnsemble_VarBC_v8.1.yaml:1]: Something minor, but now that this PR changes our built code to v8.1, I think this scenario can remove '_v8.1' from the name

File: scenarios/3dhybrid_O30kmIE60km_SpecifiedEnsemble_VarBC_v8.1.yaml
Comment (ibanos90) [inline scenarios/3dhybrid_O30kmIE60km_SpecifiedEnsemble_VarBC_v8.1.yaml:1]: My guess is that we may not need to include this scenario anymore, right?

**Comment by ibanos90** [comment]
Run the tests I got the following issue:
- HofX fails for the tests **4denvar_OIE120km_WarmStart_TEST** and **3denvar_OIE120km_IAU_WarmStart_TEST**. 
```
mpasjedi_vc_model2geovars::changevar: 'the "USGS" CRTM land surface classification 'for CRTM ObsOperators associated with infrared instruments '(i.e., obs operator.options.IRVISlandCoeff) 'can only be used with mminlu=USGS, not mminlu=MODIFIED_IGBP_MODIS_NOAH
```
- `VerifyModeBG` tasks are failing, need to update the path for the graphics

**Comment by mos3r3n** [comment]
> Run the tests I got the following issue:
> 
> * HofX fails for the tests **4denvar_OIE120km_WarmStart_TEST** and **3denvar_OIE120km_IAU_WarmStart_TEST**.
> 
> ```
> mpasjedi_vc_model2geovars::changevar: 'the "USGS" CRTM land surface classification 'for CRTM ObsOperators associated with infrared instruments '(i.e., obs operator.options.IRVISlandCoeff) 'can only be used with mminlu=USGS, not mminlu=MODIFIED_IGBP_MODIS_NOAH
> ```
> 
> * `VerifyModeBG` tasks are failing, need to update the path for the graphics

I will use Jake's 3.0.0 build and the graphic package inside alter.

I didn't test 4denvar before. Maybe need to modify the yaml for proper CRTM setting. What do you think about?

File: scenarios/3dhybrid_O60-3kmIE60km_SpecifiedEnsemble_VarBC_v8.1.yaml
Comment (mos3r3n) [reply scenarios/3dhybrid_O60-3kmIE60km_SpecifiedEnsemble_VarBC_v8.1.yaml:1]: I agree.

File: test/testinput/3denvar_O30kmIE60km_WarmStart.yaml (hunk 3 above)
Comment (mos3r3n) [reply test/testinput/3denvar_O30kmIE60km_WarmStart.yaml]: Agree.

File: initialize/applications/InitIC.py (hunk 1 above)
Comment (mos3r3n) [reply initialize/applications/InitIC.py]: I noticed that and will delete this print.

**Comment by mos3r3n** [comment]
> Run the tests I got the following issue:
> 
> * HofX fails for the tests **4denvar_OIE120km_WarmStart_TEST** and **3denvar_OIE120km_IAU_WarmStart_TEST**.
> 
> ```
> mpasjedi_vc_model2geovars::changevar: 'the "USGS" CRTM land surface classification 'for CRTM ObsOperators associated with infrared instruments '(i.e., obs operator.options.IRVISlandCoeff) 'can only be used with mminlu=USGS, not mminlu=MODIFIED_IGBP_MODIS_NOAH
> ```
> 
> * `VerifyModeBG` tasks are failing, need to update the path for the graphics

Should we change USGS to IGBP as the default?

    'IRVISlandCoeff': ['USGS', str],

**Comment by mos3r3n** [comment]
The default of 'IRVISlandCoeff' has been set to IGBP in all applications. Also, the namelist.init_atmosphere was also modified to use the new static files instead of the old topo and land use settings.

**Comment by ibanos90** [comment]
> > Run the tests I got the following issue:
> > 
> > * HofX fails for the tests **4denvar_OIE120km_WarmStart_TEST** and **3denvar_OIE120km_IAU_WarmStart_TEST**.
> > 
> > ```
> > mpasjedi_vc_model2geovars::changevar: 'the "USGS" CRTM land surface classification 'for CRTM ObsOperators associated with infrared instruments '(i.e., obs operator.options.IRVISlandCoeff) 'can only be used with mminlu=USGS, not mminlu=MODIFIED_IGBP_MODIS_NOAH
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * `VerifyModeBG` tasks are failing, need to update the path for the graphics
> 
> I will use Jake's 3.0.0 build and the graphic package inside alter.
> 
> I didn't test 4denvar before. Maybe need to modify the yaml for proper CRTM setting. What do you think about?

I tested the HofX with IGBP and it works, so we may want to add those lines to the yaml file, such as:
```
hofx:
  IRVISlandCoeff: IGBP
```

**Comment by ibanos90** [comment]
> > Run the tests I got the following issue:
> > 
> > * HofX fails for the tests **4denvar_OIE120km_WarmStart_TEST** and **3denvar_OIE120km_IAU_WarmStart_TEST**.
> > 
> > ```
> > mpasjedi_vc_model2geovars::changevar: 'the "USGS" CRTM land surface classification 'for CRTM ObsOperators associated with infrared instruments '(i.e., obs operator.options.IRVISlandCoeff) 'can only be used with mminlu=USGS, not mminlu=MODIFIED_IGBP_MODIS_NOAH
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * `VerifyModeBG` tasks are failing, need to update the path for the graphics
> 
> Should we change USGS to IGBP as the default?
> 
> ```
> 'IRVISlandCoeff': ['USGS', str],
> ```

Yes, here may be better.

**Comment by mos3r3n** [comment]
> > > Run the tests I got the following issue:
> > > 
> > > * HofX fails for the tests **4denvar_OIE120km_WarmStart_TEST** and **3denvar_OIE120km_IAU_WarmStart_TEST**.
> > > 
> > > ```
> > > mpasjedi_vc_model2geovars::changevar: 'the "USGS" CRTM land surface classification 'for CRTM ObsOperators associated with infrared instruments '(i.e., obs operator.options.IRVISlandCoeff) 'can only be used with mminlu=USGS, not mminlu=MODIFIED_IGBP_MODIS_NOAH
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > * `VerifyModeBG` tasks are failing, need to update the path for the graphics
> > 
> > 
> > Should we change USGS to IGBP as the default?
> > ```
> > 'IRVISlandCoeff': ['USGS', str],
> > ```
> 
> Yes, here may be better.

I already modified it. I didn't change that in scenarios/3dhybrid_O30kmIE60km_SpecifiedEnsemble_VarBC_allConfig.yaml because it uses the old ic and first guess file. I may modify it to use the latest files this night.

Also, I have changed the mpas-bundle to the 3.0.0 build from Jake. In case it may not fully copied, we need to wait for a while and then do tests.

**Review by ibanos90 — APPROVED** [review]
After the changes, all tests are passing now! Thanks a lot for your extensive work on this, @mos3r3n!
The '3dvar_OIE120km_ColdStart' test did work for me, I think you can remove the comment from the PR's test completed.

**Comment by mos3r3n** [comment]
> After the changes, all tests are passing now! Thanks a lot for your extensive work on this, @mos3r3n! The '3dvar_OIE120km_ColdStart' test did work for me, I think you can remove the comment from the PR's test completed.

Thanks! @ibanos90

**Review by liujake — APPROVED** [review]
Then, @mos3r3n you may merge it now.

**Review by junmeiban — APPROVED** [review]

**Pull request merged**